application/controllers/PemasokController.php

```diff
@@ -69,7 +68,6 @@ function index_put()
         $this->form_validation->set_rules('alamat', 'Alamat', 'required');
```

Comment:
yang bikin fail karena `form_validation` secara defaul tidak menerima `$this.->put()` karena `REST` Server Ci itu bikinan orang bukan dari ci official mungkin bisa dicoba `$this->form_validation->set_rules($this->pus('alamat'), 'Alamat', 'required');`